actions/lib/actions.py

```diff
@@ -104,6 +104,36 @@ def get_node(self, node):
 
         return orion_node
 
+    def get_agent(self, orion_node):
+        """
+        Get orion agent information and add it to the OrionNode object.
+        """
+        swql = """SELECT AgentID, Uri
+        FROM Orion.AgentManagement.Agent
+        WHERE NodeId=@query_on"""
+        kargs = {'query_on': orion_node.npm_id}
+        data = self.query(swql, **kargs)
+
+        if 'results' not in data:
+            msg = "No results from Orion: {}".format(data)
+            self.logger.info(msg)
+            raise Exception(msg)
+
+        if len(data['results']) == 1:
+            try:
+                orion_node.agent_id = data['results'][0]['AgentID']
```

Comment:
This agent logic should really be in `get_node` and `agent` (and the other values) are populated without needing to pass around the `orion_node` object.